Source/Core/OrientedBoundingBox.js

```diff
@@ -419,6 +419,42 @@ define([
         return result;
     };
 
+    var scratchMatrix3PP = new Matrix3();
+    var scratchCartesian3PP = new Cartesian3();
+
+    /**
+     * If the given position is not already within the box, projects the given position onto the box.
+     * @param {Cartesian3} position The position being projected onto this OrientedBoundingBox.
+     * @returns {Cartesian3} A projected version of the inputted position if it was not originally within the OrientedBoundingBox.
+     */
+    OrientedBoundingBox.prototype.clampToBounds = function(position, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('position', position);
+        //>>includeEnd('debug');
+        result = position.clone(result);
+        // to avoid inverse of a zero matrix (since that doesn't exist).
+        if (this.halfAxes.equals(Matrix3.ZERO)) {
+            result.x = 0.0;
+            result.y = 0.0;
+            result.z = 0.0;
+            return result;
+        }
+
+        // convert world space positionToCheck to orientedBoundingBox's object space.
+        var inverseTransformationMatrix = Matrix3.inverse(this.halfAxes, scratchMatrix3PP);
+        var positionInObjectSpace = Matrix3.multiplyByVector(inverseTransformationMatrix, result, scratchCartesian3PP);
+
+        // once in object space just translate to center and check if converted location is within axis oriented 2x2x2 cube (bc half axes)
+        Cartesian3.subtract(positionInObjectSpace, this.center, result);
+        result.x = CesiumMath.clamp(result.x, -1.0, 1.0);
+        result.y = CesiumMath.clamp(result.y, -1.0, 1.0);
+        result.z = CesiumMath.clamp(result.z, -1.0, 1.0);
+        Cartesian3.add(result, this.center, result);
+
+        Matrix3.multiplyByVector(this.halfAxes, result, result);
```

Comment:
I think the order of these last lines should be swapped.

Get the position in object space -> Multiply by halfAxes -> Add new relative vector to the center.